osf/models/user.py

```diff
@@ -576,7 +576,9 @@ def visible_contributor_to(self):
         """
         Nodes where user is a bibliographic contributor (group membership not factored in)
         """
-        return self.nodes.filter(is_deleted=False, contributor__visible=True, type__in=['osf.node', 'osf.registration'])
+        return self.nodes.annotate(
+            self_is_visible=Exists(Contributor.objects.filter(node_id=OuterRef('id'), user_id=self.id, visible=True))
+        ).filter(deleted__isnull=True, self_is_visible=True, type__in=['osf.node', 'osf.registration'])
```

Comment:
* Due to upgrade, the old query no longer works as expected. It doesn't tie `contributor` to `self` when checking `visible`. The following initial fix is too costly with double `INNER JOIN`

```python
return self.nodes.filter(is_deleted=False, contributor__visible=True, contributor__user=self, type__in=['osf.node', 'osf.registration'])
```

```SQL
SELECT ...
FROM "osf_abstractnode"
         INNER JOIN "osf_contributor" ON ("osf_abstractnode"."id" = "osf_contributor"."node_id")
         INNER JOIN "osf_contributor" T4 ON ("osf_abstractnode"."id" = T4."node_id")
WHERE ("osf_contributor"."user_id" = 1 AND T4."visible")
```

* Here is the query with this updated fix.

```SQL
SELECT ...
       EXISTS(SELECT (1) AS "a"
              FROM "osf_contributor" U0
              WHERE (U0."node_id" = "osf_abstractnode"."id" AND U0."user_id" = 1 AND U0."visible")
              LIMIT 1) AS "self_is_visible"
FROM "osf_abstractnode"
         INNER JOIN "osf_contributor" ON ("osf_abstractnode"."id" = "osf_contributor"."node_id")
WHERE ("osf_contributor"."user_id" = 1 AND EXISTS(SELECT (1) AS "a"
                                                  FROM "osf_contributor" U0
                                                  WHERE (U0."node_id" = "osf_abstractnode"."id" AND U0."user_id" = 1 AND
                                                         U0."visible")
                                                  LIMIT 1))
```